### DISCUSSION TITLE
in channel_post_handler no information from senders

### DISCUSSION THREAD

**Discussion by Rilin** [eternnoir/pyTelegramBotAPI#2002] [Q&A]
Please answer these questions before submitting your issue. Thanks!

1. What version of pyTelegramBotAPI are you using?
4.12.0

2. What OS are you using?
 Windows10/"Oracle Linux Server" VERSION="8.6"

3. What version of python are you using?
Python 3.10.5

in @tb.channel_post_handler(content_types = ['text'])
def echo_all(message):
print(message)

not information from user sender.

message send from android phone via sms - > share -> telegram -> channel

{
    'content_type': 'text', 'id': 2949, 'message_id': 2949, 'from_user': None, 'date': 1687890257, 
    'chat': {
        'id': -10018324xxxxx, 'type': 'channel', 'title': 'xxxBot', 'username': None, 'first_name': None, 'last_name': None, 'is_forum': None, 'photo': None, 'bio': None, 
        'join_to_send_messages': None, 'join_by_request': None, 'has_private_forwards': None, 'has_restricted_voice_and_video_messages': None, 'description': None, 
        'invite_link': None, 'pinned_message': None, 'permissions': None, 'slow_mode_delay': None, 'message_auto_delete_time': None, 'has_protected_content': None, 
        'sticker_set_name': None, 'can_set_sticker_set': None, 'linked_chat_id': None, 'location': None, 'active_usernames': None, 'emoji_status_custom_emoji_id': None, 
        'has_hidden_members': None, 'has_aggressive_anti_spam_enabled': None
        }, 
    'sender_chat': {
        'id': -10018324xxxxx, 'type': 'channel', 'title': xxxBot', 'username': None, 'first_name': None, 'last_name': None, 'is_forum': None, 'photo': None, 'bio': None, 
        'join_to_send_messages': None, 'join_by_request': None, 'has_private_forwards': None, 'has_restricted_voice_and_video_messages': None, 'description': None, 'invite_link': None, 
        'pinned_message': None, 'permissions': None, 'slow_mode_delay': None, 'message_auto_delete_time': None, 'has_protected_content': None, 'sticker_set_name': None, 
        'can_set_sticker_set': None, 'linked_chat_id': None, 'location': None, 'active_usernames': None, 'emoji_status_custom_emoji_id': None, 'has_hidden_members': None, 
        'has_aggressive_anti_spam_enabled': None
        }, 
    'forward_from': None, 'forward_from_chat': None, 'forward_from_message_id': None, 'forward_signature': None, 'forward_sender_name': None, 'forward_date': None, 
    'is_automatic_forward': None, 'reply_to_message': None, 'via_bot': None, 'edit_date': None, 'has_protected_content': None, 'media_group_id': None, 
    'author_signature': None, 
    'text': 'xxxxxxxxxxxxxxxxxxxxxxxxxx', 
    'entities': None, 'caption_entities': None, 'audio': None, 'document': None, 'photo': None, 'sticker': None, 'video': None, 'video_note': None, 'voice': None, 'caption': None, 
    'contact': None, 'location': None, 'venue': None, 'animation': None, 'dice': None, 'new_chat_member': None, 'new_chat_members': None, 'left_chat_member': None, 
    'new_chat_title': None, 'new_chat_photo': None, 'delete_chat_photo': None, 'group_chat_created': None, 'supergroup_chat_created': None, 'channel_chat_created': None, 
    'migrate_to_chat_id': None, 'migrate_from_chat_id': None, 'pinned_message': None, 'invoice': None, 'successful_payment': None, 'connected_website': None, 
    'reply_markup': None, 'message_thread_id': None, 'is_topic_message': None, 'forum_topic_created': None, 'forum_topic_closed': None, 'forum_topic_reopened': None, 
    'has_media_spoiler': None, 'forum_topic_edited': None, 'general_forum_topic_hidden': None, 'general_forum_topic_unhidden': None, 
    'write_access_allowed': None, 'user_shared': None, 'chat_shared': None, 
    'json': {
        'message_id': 2949, 
        'sender_chat': {'id': -10018324xxxxx, 'title': 'xxxBot', 'type': 'channel'}, 
        'chat': {'id': -10018324xxxxx, 'title': xxxBot', 'type': 'channel'}, 
        'date': 1687890257, 'text': 'xxxxxxxxxxxxxxxxxxxxxxxxxx'
        }
    }

**Comment by Badiboy** [comment]
> no information from senders

Yes. Channel is anonymous.

**Reply by hrfahimi** [reply]
I want to send messages(any kind) from private channel that bot is admin to users.
I use "**channel_post_handler**" as decorator and "**forward_message**" for send **any kind**(text, photo, voice, ....), but it's not possible to send without "from_chat_id" and **hide sender** ('Forwarded from ....' section)!!
any suggestions?

**Reply by hrfahimi** [reply]
I found solution from telegram group: **copy_message**!
But I think "copy" is not good naming :))

**Reply by coder2020official** [reply]
We don't decide the namings, it is telegram API
